### PR TITLE
Power Word Radiance module - disc priest

### PR DESCRIPTION
**Pull request by oisinod**
this will be used in the guide component

### CODE REVIEW

File: src/analysis/retail/priest/discipline/modules/spells/PowerWordRadiance.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    if (this.radianceCasts.length === 0) {
+      return;
+    }
+
+    if (event.timestamp === this.radianceCasts[this.radianceCasts.length - 1].cast.timestamp) {
+      this.radAtones += 1;
+      if (this.radAtones === 5) {
+        this.radAtones = 0;
+        this.radianceCasts[this.radianceCasts.length - 1].goodCast = true;
+        return;
```
Comment (danny-janse) [inline src/analysis/retail/priest/discipline/modules/spells/PowerWordRadiance.tsx]: This return is unnecessary.

File: src/analysis/retail/priest/discipline/modules/spells/PowerWordRadiance.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  goodCast: boolean;
+  onAtoned: boolean;
+};
+
+class PowerWordRadiance extends Analyzer {
+  radiancesOnAtonedTarget = 0;
+  atonementsPreRadiance = 0;
+  lastPWR: CastEvent | null = null;
+  radianceCasts: RadianceInfo[] = [];
+  radAtones = 0;
```
Comment (danny-janse) [inline src/analysis/retail/priest/discipline/modules/spells/PowerWordRadiance.tsx:16]: Could you clear up the extra whitespaces?

File: src/analysis/retail/priest/discipline/modules/spells/PowerWordRadiance.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+type RadianceInfo = {
+  cast: CastEvent;
+  goodCast: boolean;
+  onAtoned: boolean;
+};
+
+class PowerWordRadiance extends Analyzer {
+  radiancesOnAtonedTarget = 0;
+  atonementsPreRadiance = 0;
+  lastPWR: CastEvent | null = null;
```
Comment (danny-janse) [inline src/analysis/retail/priest/discipline/modules/spells/PowerWordRadiance.tsx]: The above three properties seem to be unused. If you need to access them for the guide i'd suggest you store them together with the other data in `RadianceInfo`.

File: src/analysis/retail/priest/discipline/modules/spells/PowerWordRadiance.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  static dependencies = {
+    combatants: Combatants,
+    atonement: Atonement,
+  };
+  protected atonement!: Atonement;
+  protected combatants!: Combatants;
+
+  constructor(options: Options) {
+    super(options);
+
```
Comment (danny-janse) [inline src/analysis/retail/priest/discipline/modules/spells/PowerWordRadiance.tsx:26]: This module should only be active if the talent is chosen.

**Review by danny-janse — CHANGES_REQUESTED** [review]
Althought a bit confusing this seems to be working correctly.

**Comment by oisinod** [comment]
@djanse comments addressed :)

File: src/analysis/retail/priest/discipline/modules/spells/PowerWordRadiance.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import Atonement from './Atonement';
+
+type RadianceInfo = {
+  cast: CastEvent;
+  goodCast: boolean;
+  onAtoned: boolean;
+};
+
+class PowerWordRadiance extends Analyzer {
+  radiancesOnAtonedTarget = 0;
```
Comment (danny-janse) [inline src/analysis/retail/priest/discipline/modules/spells/PowerWordRadiance.tsx]: Correct me if im wrong but i believe `radiancesOnAtonedTarget` can go too?

File: src/analysis/retail/priest/discipline/modules/spells/PowerWordRadiance.tsx (hunk 4 above)
Comment (danny-janse) [reply src/analysis/retail/priest/discipline/modules/spells/PowerWordRadiance.tsx:26]: Could you still add an active check to the module?
```tsx
this.active = this.selectedCombatant.hasTalent(...);

if (!this.active) {
  return;
}
```

**Review by danny-janse — CHANGES_REQUESTED** [review]
Looks alot better :) Still a few changes that i'd like you to make.

File: src/analysis/retail/priest/discipline/modules/spells/PowerWordRadiance.tsx (hunk 5 above)
Comment (oisinod) [reply src/analysis/retail/priest/discipline/modules/spells/PowerWordRadiance.tsx]: 👍

**Comment by oisinod** [comment]
I didn't add the active thing because I thought it was a mandatory talent but you actually can avoid it even though it's on the second row 😆

**Review by danny-janse — APPROVED** [review]
Looks good now.

**Pull request merged**